File: go/pig-latin/pig_latin.go
package piglatin

import "strings"

func Sentence(sentence string) string {
	splitSentence := strings.Split(sentence, " ")
	words := []string{}
	for _, word := range splitSentence {
		pos := 0
		for pos = 0; pos < len(word); pos++ {
			if pos == 0 && len(word) > 2 && (word[0:2] == "xr" || word[0:2] == "yt") {
				break
			} else if word[pos] == 'q' && pos < len(word)-1 && word[pos+1] == 'u' {
				pos += 2
			} else if pos > 0 && word[pos] == 'y' {
				break
			} else if word[pos] != 'a' && word[pos] != 'e' && word[pos] != 'i' && word[pos] != 'o' && word[pos] != 'u' {
				continue
			}
			break
		}
		words = append(words, word[pos:]+word[:pos]+"ay")
	}
	return strings.Join(words, " ")
}

File: python/pig-latin/pig_latin.py
def translate(text):
    words = []
    for word in text.split():
        if len(word) >= 2 and word[0:2] in ["xr", "yt"]:
            words.append(word + "ay")
            continue
        for pos in range(len(word)):
            if word[pos] == "q" and pos < len(word) - 1 and word[pos + 1] == "u":
                pos += 2
            elif word[pos] not in ["a", "e", "i", "o", "u"] and (pos == 0 or word[pos] != "y"):
                continue
            break
        words.append(word[pos:] + word[:pos] + "ay")
    return " ".join(words)
